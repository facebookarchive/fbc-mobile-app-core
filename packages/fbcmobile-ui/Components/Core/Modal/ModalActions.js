/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';

import * as React from 'react';
import Button from '@fbcmobile/ui/Components/Core/Button';
import Divider from '@fbcmobile/ui/Components/Core/Divider';
import Text from '@fbcmobile/ui/Components/Core/Text';
import {StyleSheet, TouchableOpacity, View} from 'react-native';

type Props = {
  +cancelLabel?: React.Node,
  +okLabel: React.Node,
  +onCancelPressed?: () => void,
  +onOkPressed: () => void | Promise<void>,
  +isOkDisabled?: boolean,
  +showDivider?: boolean,
  +style?: ViewStyleProp,
};

const ModalActions = ({
  cancelLabel,
  okLabel,
  onCancelPressed,
  onOkPressed,
  isOkDisabled = false,
  showDivider,
  style,
}: Props) => {
  return (
    <View style={style}>
      {showDivider ? <Divider /> : null}
      <View style={styles.buttonsContainer}>
        {cancelLabel != null ? (
          <View style={styles.cancelButton}>
            <TouchableOpacity style={styles.link} onPress={onCancelPressed}>
              <Text color="primary" weight="bold">
                {cancelLabel}
              </Text>
            </TouchableOpacity>
          </View>
        ) : null}
        <View style={styles.gap} />
        <View style={styles.okButton}>
          <Button
            variant="primary"
            onPress={onOkPressed}
            disabled={isOkDisabled}>
            {okLabel}
          </Button>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  buttonsContainer: {
    display: 'flex',
    flexDirection: 'row',
    paddingHorizontal: 15,
    paddingVertical: 10,
  },
  gap: {
    flexGrow: 1,
  },
  cancelButton: {
    alignSelf: 'flex-start',
  },
  okButton: {
    alignSelf: 'flex-end',
  },
  link: {
    paddingVertical: 10.5,
    paddingHorizontal: 19,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default ModalActions;
