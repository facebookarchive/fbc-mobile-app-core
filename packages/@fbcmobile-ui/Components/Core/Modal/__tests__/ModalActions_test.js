/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import ModalActions from '../ModalActions';
import React from 'react';
import renderer from 'react-test-renderer';

describe('ModalActions Tests', () => {
  test('ModalActions renders correctly', () => {
    const tree = renderer
      .create(<ModalActions okLabel={'ok'} onOkPressed={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ModalActions renders correctly with divider', () => {
    const tree = renderer
      .create(
        <ModalActions
          okLabel={'ok'}
          onOkPressed={jest.fn()}
          showDivider={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ModalActions renders correctly with cancel label', () => {
    const tree = renderer
      .create(
        <ModalActions
          okLabel={'ok'}
          onOkPressed={jest.fn()}
          cancelLabel={'cancel'}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });});
