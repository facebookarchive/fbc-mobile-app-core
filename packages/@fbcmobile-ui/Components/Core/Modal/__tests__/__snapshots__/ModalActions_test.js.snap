// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalActions Tests ModalActions renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1877F2",
            "borderRadius": 4,
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 10.5,
          }
        }
      >
        <Text
          style={
            Array [
              Array [
                Object {
                  "color": "#484848",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "lineHeight": 20,
                },
                null,
                Object {
                  "fontFamily": "Roboto-Bold",
                },
                Object {
                  "textAlign": "left",
                },
              ],
              Array [
                Object {
                  "display": "flex",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                Object {
                  "color": "white",
                },
                Object {
                  "fontSize": 16,
                },
                null,
                Object {
                  "textAlign": "center",
                },
              ],
            ]
          }
        >
          ok
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ModalActions Tests ModalActions renders correctly with cancel label 1`] = `
<View>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 19,
            "paddingVertical": 10.5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#484848",
                "fontSize": 16,
                "fontWeight": "500",
                "lineHeight": 20,
              },
              Object {
                "color": "#1877F2",
              },
              Object {
                "fontFamily": "Roboto-Bold",
              },
              Object {
                "textAlign": "left",
              },
            ]
          }
        >
          cancel
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1877F2",
            "borderRadius": 4,
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 10.5,
          }
        }
      >
        <Text
          style={
            Array [
              Array [
                Object {
                  "color": "#484848",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "lineHeight": 20,
                },
                null,
                Object {
                  "fontFamily": "Roboto-Bold",
                },
                Object {
                  "textAlign": "left",
                },
              ],
              Array [
                Object {
                  "display": "flex",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                Object {
                  "color": "white",
                },
                Object {
                  "fontSize": 16,
                },
                null,
                Object {
                  "textAlign": "center",
                },
              ],
            ]
          }
        >
          ok
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ModalActions Tests ModalActions renders correctly with divider 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CFD1D5",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1877F2",
            "borderRadius": 4,
            "display": "flex",
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 10.5,
          }
        }
      >
        <Text
          style={
            Array [
              Array [
                Object {
                  "color": "#484848",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "lineHeight": 20,
                },
                null,
                Object {
                  "fontFamily": "Roboto-Bold",
                },
                Object {
                  "textAlign": "left",
                },
              ],
              Array [
                Object {
                  "display": "flex",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                Object {
                  "color": "white",
                },
                Object {
                  "fontSize": 16,
                },
                null,
                Object {
                  "textAlign": "center",
                },
              ],
            ]
          }
        >
          ok
        </Text>
      </View>
    </View>
  </View>
</View>
`;
